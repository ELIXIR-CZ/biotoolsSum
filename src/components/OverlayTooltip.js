import * as React from 'react'
import { OverlayTrigger, Tooltip } from 'react-bootstrap'

export const OverlayTooltip = ({ id, tooltipText, children }) => {
  const tooltip = <Tooltip id={id}>{tooltipText}</Tooltip>

  return (
    <OverlayTrigger
      overlay={tooltip} placement='top'
      delayShow={300} delayHide={150}
      >
      {children}
    </OverlayTrigger>
  )
}
