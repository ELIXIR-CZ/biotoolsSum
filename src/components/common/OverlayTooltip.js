import React from 'react'
import { OverlayTrigger, Tooltip } from 'react-bootstrap'

export const OverlayTooltip = ({ id, tooltipText, children, delayShow, delayHide, placement }) => {
  const tooltip = <Tooltip id={id}>{tooltipText}</Tooltip>

  return (
    <OverlayTrigger
      overlay={tooltip}
      placement={placement}
      delayShow={delayShow}
      delayHide={delayHide}
    >
      {children}
    </OverlayTrigger>
  )
}

OverlayTooltip.defaultProps = {
  delayShow: 300,
  delayHide: 150,
  placement: 'top',
}
