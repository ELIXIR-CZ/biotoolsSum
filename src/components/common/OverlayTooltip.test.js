import React from 'react'
import renderer from 'react-test-renderer'
import OverlayTooltip from './OverlayTooltip'

it('Renders OverlayTooltip correctly', () => {
  const tree = renderer
    .create(
      <OverlayTooltip id='overlay_snapshot' tooltipText='This is overlay snapshot'>
        <span>{'OverlayTooltip child'}</span>
      </OverlayTooltip>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
