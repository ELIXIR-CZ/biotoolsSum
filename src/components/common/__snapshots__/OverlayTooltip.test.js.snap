exports[`test Renders OverlayTooltip correctly 1`] = `
<span
  onBlur={[Function]}
  onClick={null}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}>
  OverlayTooltip child
</span>
`;
